### DISCUSSION TITLE
Kernel list does not update

### DISCUSSION THREAD

**Discussion by RalfKliemt** [microsoft/vscode-jupyter#11017] [Questions and Answers] [answered]
Hello,

I am trying to make a custom Kernel available in VSCode. It does not appear in the list of Kernels. 
However, some older partly custom, partly even deleted Kernel settings (.json) are there. Reloading, quitting VSCode does not help. 
In between I once had an update to VSCode and it seems then the kernel lists was reloaded.
Is there a command or button to reload the Kernel list manually?

Cheers!

  BTW: in a Jupyter Notebook this is no issue.

**Comment by DonJayamanne** [comment, marked as answer]
Thanks for filing this issue, please could you let us know where this custom kernel is defined on your machine and also provide the contents of the kernelspec.json file. thanks.
Also, please could you:
* Open the user settings file and add the following in there: `"jupyter.logging.level": "verbose",`
* Next, reload VS Code
* Open a notebook, wait for a few seconds
* Then capture all of the output from the `Jupyter` output panel (go into `Jupyter: Show Output`) and paste all of that here.

THe logs show where we look for kernels and the list of kernels displayed (& optionally filtered & why).

**Reply by RalfKliemt** [reply]
Thanks for the quick reply. 

- The increased logging level showed me what went wrong
- Basically I expected to have a python executable available like in my shell. VSCode however, does not read from `.bash_profile`. 
- VSCode does not list kernels which it cannot load. 
- I solved the issue by using the full path to the python executable
- 
**My suggestion** would be to list non-functional kernels with red colour to indicate that there are problems.

Cheers!
